### PR TITLE
Uplift master branch to match the latest state of ocl-open-70

### PR DESCRIPTION
**Pull request by AlexeySachkov**
I squashed some commits to reduce their number

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    |   `-- clang
+    `-- projects
+        |-- llvm-spirv
+        `-- common-clang
+```
+
+This can be done using the following commands:
+```
+cd <workspace>
+git clone https://github.com/llvm-mirror/llvm.git -b release_70
```
Comment (AlexeySotkin) [inline README.md]: Please drop " -b release_70" in git commands here and below.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+and SPIR-V Translator libraries. **Note:** currently this kind of build is
+supported on Linux only.
+
+Integration with pre-built LLVM is done using standard `find_package` way as
+documented in [Embedding LLVM in your project](https://llvm.org/docs/CMake.html#embedding-llvm-in-your-project).
+
+Commands to checkout sources and build:
+```
+cd <workspace>
+git clone https://github.com/intel/opencl-clang.git -b ocl-open-70
```
Comment (AlexeySotkin) [inline README.md]: Please drop " -b ocl-open-70"

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,306 +1,291 @@
...
+
+include(CMakeFunctions)
+
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    set(USE_PREBUILT_LLVM ON)
+
+    add_definitions(-DUSE_PREBUILT_LLVM)
+
+    if(NOT PREFERRED_LLVM_VERSION)
+        set(PREFERRED_LLVM_VERSION "7.0.0")
```
Comment (AlexeySotkin) [inline CMakeLists.txt]: Please update to llvm 8

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,306 +1,291 @@
...
+endif (WIN32)
+
+# set that name of the main output file as a target name
+if (NOT DEFINED COMMON_CLANG_LIBRARY_NAME)
+    set(COMMON_CLANG_LIBRARY_NAME common_clang)
 endif()
+set(TARGET_NAME ${COMMON_CLANG_LIBRARY_NAME}${BUILD_PLATFORM} )
+
+if(NOT USE_PREBUILT_LLVM)
+    set(TARGET_BRANCH "ocl-open-70")
```
Comment (AlexeySotkin) [inline CMakeLists.txt]: Should we change it to  "ocl-open-80" ?

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,306 +1,291 @@
...
+    set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/tools/clang)
+    set(CLANG_BASE_REVISION e44b4e4240e03c24f5479f81235519493823c27b)
```
Comment (AlexeySotkin) [inline CMakeLists.txt]: I think we can set it to "master"

File: CMakeLists.txt [hunk 6]
```diff
@@ -1,306 +1,291 @@
...
+
+    set(SPIRV_SOURCE_DIR ${LLVM_SOURCE_DIR}/projects/llvm-spirv)
+    set(SPIRV_BASE_REVISION 4d62009e2225024abd481ca982ec3d63304df3f0)
```
Comment (AlexeySotkin) [inline CMakeLists.txt]: I think we can set it to "master"

File: README.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+mkdir build && cd build
+cmake ../opencl-clang
+make all -j`nproc`
+```
+
+#### Configuration options
+
+##### Preferred LLVM version
+
+By default, Common clang's cmake script is searching for LLVM 7.0.0. You can
```
Comment (AlexeySotkin) [inline README.md]: On the master branch we should search for the latest LLVM.

File: README.md (hunk 2 above)
Comment (AlexeySachkov) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (AlexeySachkov) [reply README.md]: Done

File: CMakeLists.txt (hunk 6 above)
Comment (AlexeySachkov) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 5 above)
Comment (AlexeySachkov) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 4 above)
Comment (AlexeySachkov) [reply CMakeLists.txt]: Done

File: README.md (hunk 1 above)
Comment (AlexeySachkov) [reply README.md]: Done

File: CMakeLists.txt (hunk 3 above)
Comment (AlexeySachkov) [reply CMakeLists.txt]: Done

**Pull request merged**